app/scripts/services/spouse.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+angular.module('confRegistrationWebApp')
+  .factory('spouse', function ($http, $cacheFactory) {
+    var spouseRegistrationCache = $cacheFactory('spouseRegistration');
+
+    var spouse = {
+      // Preload a spouse registration so that calling getSpouseRegistration for that conference will complete sooner
+      // This functionality is provided because looking up a spouse registration via getSpouseRegistration is a lengthy operation
+      preloadSpouseRegistration: function (conferenceId) {
+        return spouse.getSpouseRegistration(conferenceId);
+      },
+
+      // Lookup the registration for the logged in user's spouse for a particular conference
+      // Return a promise that resolves to the registration if it could be found, or null if it could not
+      getSpouseRegistration: function (conferenceId) {
+        return $http.get('conferences/' + conferenceId + '/registrations/spouse', {
+          cache: spouseRegistrationCache
+        }).then(function (res) {
+          if (res.status === 200) {
+            // The spouse registration was found
+            return res.data;
+          }
+
+          // The spouse registration was not found
+          return null;
+        }).catch(function () {
```

Comment:
I guess I'm usually not doing anything in the success that I'd want caught by the error handler. I guess I normally use the error handler to tell the user there was an api error. If you run into that case, yes you'd want a separate error handler. Otherwise I don't think it matters much. RxJS uses subscribe(success,error,done) which I guess is what I've gotten used to.